### PR TITLE
Add Prometheus configuration and recording rules for DSP Operator

### PR DESCRIPTION
**Pull request by DharmitD**
## Description

Adding Prometheus configuration and recording rules for Data Science Pipelines' Operator

## How Has This Been Tested?

On PSI/OSD with Olminstall:

- Using olminstall, trigger first a clean build with the image `quay.io/hukhan/rhods-operator-live-catalog:1.25.0-dsp-rules-1-25`. Make sure that the installation is successful
- Go to the prometheus instance url in the `redhat-ods-monitoring` namespace, query the recording rules, make sure that they return entries and values relevant to the data science pipelines' operator

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message.
- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-7876
- [x] The Jira story is acked.
- [x] Live build image: quay.io/hukhan/rhods-operator-live-catalog:1.25.0-dsp-rules-1-25
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [ ] QE contact acknowledges that this has been tested and is approved for merge.

### CODE REVIEW

File: monitoring/prometheus/prometheus-configs.yaml [hunk 1]
```diff
@@ -669,6 +760,27 @@ data:
...
+          namespaces:
+            names:
+              - redhat-ods-applications
+      relabel_configs:
+        - source_labels: [__address__]
+          regex: (.+):(\d+)
+          target_label: __address__
+          replacement: ${1}:8080
+        - source_labels: [__meta_kubernetes_service_name]
+          regex: ^(data-science-pipelines-operator)$
```
Comment (harshad16) [inline monitoring/prometheus/prometheus-configs.yaml:777]:
```suggestion
          regex: ^(data-science-pipelines-operator-service)$
```

**Review by harshad16 — COMMENTED** [review]
few doubts

**Comment by tarukumar** [comment]
@jgarciao Kindly look into it

**Review by harshad16 — COMMENTED** [review]
/lgtm

:+1:

**Comment by jgarciao** [comment]
I've tested livebuild http://quay.io/rhn_support_ddalvi/rhods-operator-live-catalog:1.25.0-rhods-7501:

- Prometheus > Status > Targets > user_facing_endpoints_status: contains endpoint for data-science-pipelines-operator
- Prometheus > Graphs > `{job="user_facing_endpoints_status", name="data-science-pipelines-operator"}` shows some metrics , but **I believe they are not correct (info below)**

For example, metric `probe_success{instance="data-science-pipelines-operator-service.redhat-ods-applications.svc:8080/metrics", job="user_facing_endpoints_status", name="data-science-pipelines-operator"}` seems to return always value 0, independently of the number of pods running in deployment data-science-pipelines-operator-controller-manager. 

On the contrary, the probe_success metric for odh-notebook-controller-service value is 1 and, when scaling the Deployment to zero pods, it changes to zero (see screenshot below)

livebuild-probe-success-data-science-pipelines

The Service name for data-science-pipelines-operator seems wrong in RHODS 1.25 RC1

rhods-1 25-RC1-dspo-service
rhods-1 25-RC1-services

**Comment by jgarciao** [comment]
With the livebuild at http://quay.io/hukhan/rhods-operator-live-catalog:1.25.0-dsp-rules-1-25 I've verified that the probe_success metric works well for data-science-pipelines-operator (it detects when it's working and when it's not)

I think the PR can be merged

livebuild-PR413-data-science-pipelines-operator-probe-success-metric

**Review by jgarciao — APPROVED** [review]
lgtm

**Pull request merged**